updater/update/update.go

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2021 PaperCut Software International Pty. Ltd.
+ */
+
+package update
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+type UpgradeInfo struct {
+	URL        string
+	Version    string
+	Sha1       string
+	Sha256     string
+	Operations []Operation
+}
+
+type Operation struct {
+	Action string
+	Args   []string
+}
+
+func Check(url string, currentVer string) (*UpgradeInfo, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", url+"?version="+currentVer, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("User-Agent", "Update Check")
```

Comment:
NIT: Should we add some kind of version to help us identify the updater in the future?

Reply:
yep I do agree the version could be here...but not trying to solve every problem, just splitting out the updater out of gigantic single file CLI